src/containers/SearchContainer/index.js

```diff
@@ -0,0 +1,213 @@
+//       score: 42.34,
+//       matchData: {
+//         metadata: {
+//           <term1>: {
+//             <doc_field1>: { position: [ [<start>, <length>], ... ] },
+//             <doc_field2>: { position: [ [<start>, <length>], ... ] },
+//             ...
+//           },
+//           <term2>: {
+//             <doc_field1>: { position: [ [<start>, <length>], ... ] },
+//             <doc_field2>: { position: [ [<start>, <length>], ... ] },
+//             ...
+//           },
+//         },
+//       },
+//     },
+//     ...
+//   ]
+const getPositions = (searchResult, field) => {
+  let positions = [];
+  if (searchResult.matchData && searchResult.matchData.metadata) {
+    const data = searchResult.matchData.metadata;
+    Object.keys(data).forEach(searchTerm => {
+      if (data[searchTerm][field] && data[searchTerm][field].position) {
+        positions = positions.concat(data[searchTerm][field].position);
+      }
+    });
+  }
+  return positions;
+};
+
+const getSearchResults = query => {
+  let searchResults = [];
+  const categoryMap = { 'All Results': true };
+
+  if (query && window.__LUNR__ && window.__LUNR__.en) {
+    try {
+      const queryResults = window.__LUNR__.en.index.search(query);
+
+      searchResults = queryResults.map(searchResult => {
+        const doc = window.__LUNR__.en.store[searchResult.ref];
+        categoryMap[doc.sourceInstanceName] = true;
+        return {
+          title: getPositions(searchResult, 'title'),
+          body: getPositions(searchResult, 'body'),
+          tags: getPositions(searchResult, 'tags'),
+          doc,
+        };
+      });
+    } catch (x) {
+      // bad query e.g. title:
+    }
+  }
+
+  return { searchResults, searchCategories: Object.keys(categoryMap).sort() };
+};
+
+const SearchContainer = ({ location }) => {
+  const { term } = useParams(location);
+  const initialSearch = getSearchResults(term);
+  const [value, setValue] = useState(term);
+  const [results, setResults] = useState(initialSearch.searchResults);
+  const [categories, setCategories] = useState(initialSearch.searchCategories);
+  const [activeCategoryIndex, setActiveCategoryIndex] = React.useState(0);
+
+  const onChange = event => {
+    const { value: newValue } = event.target;
+    setValue(newValue);
+
+    // update the URL
+    const query = newValue ? `?term=${encodeURIComponent(newValue)}` : '';
+    navigate(`/search${query}`, { replace: true });
+
+    const { searchResults, searchCategories } = getSearchResults(newValue);
+    setResults(searchResults);
+    setCategories(searchCategories);
+    // todo update route term= param
+  };
+
+  const filterResults = category =>
+    results.filter(
+      ({ doc }) =>
+        !category ||
+        category === 'All Results' ||
+        doc.sourceInstanceName === category,
+    );
+
+  const onCategoryChange = index => setActiveCategoryIndex(index);
+
+  return (
+    <Box flex overflow="auto" gap="medium" pad="small">

```

Comment:
we'll be releasing a new component called Main on Grommet today/tomorrow, I think this Box is a good candidate to use Main instead.

Reply:
Cool. I'll watch for it.